pkg/judge/judge.go

```diff
@@ -0,0 +1,127 @@
+package judge
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/jasondellaluce/synchro/pkg/utils"
+)
+
+type ChangeType int
+
+const (
+	Modify ChangeType = iota
+	Add
+	Delete
+	Rename
+)
+
+type CommitChange struct {
+	FileName string
+	Type     ChangeType
+}
+
+func Judge(ctx context.Context, git utils.GitHelper, commit string) error {
+	out, err := git.DoOutput("show", "--raw", commit)
+	if err != nil {
+		return err
+	}
+
+	modified := 0
+	added := 0
+	deleted := 0
+	renamed := 0
+
+	lines := strings.Split(out, "\n")
+	var changes []CommitChange
+
+	for _, line := range lines {
+		if len(line) > 0 && line[0] == ':' {
+			change, err := parseMetadataInfo(line)
+			if err != nil {
+				return err
+			}
+			changes = append(changes, change)
+		}
+	}
+
+	for _, change := range changes {
+		switch change.Type {
+		case Modify:
+			modified = 1
+		case Add:
+			added = 1
+		case Delete:
+			deleted = 1
+		case Rename:
+			renamed = 1
+		}
+	}
+
+	if (modified + added + deleted + renamed) > 1 {
+		err = fmt.Errorf("commits can either have modified, added, deleted or renamed files and not a combination of those")
+
+		for _, change := range changes {
+			if modified != 0 && change.Type == Modify {
+				fmt.Fprintf(os.Stderr, "Modified file: %s\n", change.FileName)
```

Comment:
```suggestion
				fmt.Fprintf(os.Stdout, "Modified file: %s\n", change.FileName)
```

Let's put each of these as stdout, and also print the same message you store in `err` before printing those. That way, stderr will contain all the logged backtrack, whereas stdout will contains clear guidance about what's wrong. Automated tooling will just grab stdout.

Looks amazing after this!